packages/teams-js/src/public/video.ts

```diff
@@ -116,25 +226,121 @@ export namespace video {
    */
   export type VideoEffectCallback = (effectId: string | undefined) => Promise<void>;
 
+  type MediaStreamInfo = {
+    streamId: string;
+  };
+
   /**
    * Register to read the video frames in Permissions section
    * @beta
-   * @param frameCallback - The callback to invoke when registerForVideoFrame has completed
-   * @param config - VideoFrameConfig to customize generated video frame parameters
+   * @param option - Callbacks and configuration to process the video frames. A host may support either {@link MediaStreamCallback} or {@link SharedFrameCallback}, but not both.
+   * To ensure the video effect works on all supported hosts, the video app should provide both {@link MediaStreamCallback} and {@link SharedFrameCallback}.
+   * The host will choose the appropriate callback based on the host's capability.
+   *
+   * @example
+   * ```typescript
+   * video.registerForVideoFrame({
+   *   mediaStreamCallback: async (receivedVideoFrame) => {
+   *     const originalFrame = receivedVideoFrame.videoFrame;
+   *     try {
+   *       const processedFrame = await processFrame(originalFrame);
+   *       return processedFrame;
+   *     } catch (e) {
+   *       throw e;
+   *     }
+   *   },
+   *   sharedFrameCallback: (
+   *     frame: VideoFrameData,
+   *     notifyVideoFrameProcessed: notifyVideoFrameProcessedFunctionType,
+   *     notifyError: notifyErrorFunctionType
+   *     ) => {
+   *       try {
+   *         processFrameInplace(frame);
+   *         notifyVideoFrameProcessed();
+   *       } catch (e) {
+   *         notifyError(e);
+   *       }
+   *     },
+   *   config: {
+   *     format: video.VideoPixelFormat.NV12,
+   *   }
+   * });
+   * ```
    */
-  export function registerForVideoFrame(frameCallback: VideoFrameCallback, config: VideoFrameConfig): void {
+  export function registerForVideoFrame(option: VideoFrameCallbackOptions): void {
```

Comment:
Once you rename the "VideoFrameCallbackOptions" as per my comment above, you should probably rename the "options" to "params", "videoFrameParams", or something along those lines.

Reply:
updated